src/pages/index.js

```diff
@@ -1,90 +1,233 @@
+  handleNewCardFormSubmit
 );
-editProfilePopup.setEventListeners();
+cardPopupForm.setEventListeners();
+
+// Add New Card Form Submit Handler
+function handleNewCardFormSubmit(userInfo) {
+  cardPopupForm.setLoading(true);
+  api
+    .fetchNewCard(userInfo)
+    .then((res) => {
+      cardList.addItem(res);
+    })
+    .then(() => {
+      addCardFormElement.reset();
+      addFormValidator.disableButton();
+      cardPopupForm.close();
+    })
+    .catch(console.error)
+    .finally(() => cardPopupForm.setLoading(false));
+}
+
+// Initialize Popup With Image
+const popupWithImage = new PopupWithImage("#preview-picture-modal");
+popupWithImage.setEventListeners();
 
-// Initialize UserInfo
-const userInfo = new UserInfo(profileSelectors);
+// Handle Preview Picture
+function handlePreviewPicture(cardData) {
+  popupWithImage.open(cardData);
+}
 
-document.querySelector("#profile-edit-button").addEventListener("click", () => {
+// Initialize Profile Popup Form
+const profilePopupForm = new PopupWithForm(
+  "#profileEditModal",
+  handleProfileEditSubmit
+);
+profilePopupForm.setEventListeners();
+
+// Profile Edit Button Click Event
+profileEditButton.addEventListener("click", () => {
+  profilePopupForm.open();
   const userData = userInfo.getUserInfo();
-  editProfilePopup.setInputValues({
-    title: userData.name,
-    description: userData.job,
-  });
-  editProfilePopup.open();
+  profileTitleInput.value = userData.title;
+  profileDescriptionInput.value = userData.description;
+  editFormValidator.disableButton();
 });
 
-document.querySelector("#modal-add-button").addEventListener("click", () => {
-  addCardPopup.open();
-});
+// Add New Card Button Click Event
+addNewCardButton.addEventListener("click", () => cardPopupForm.open());
+
+// Initialize Delete Confirmation Popup
+const deleteConfirmPopup = new PopupWithConfirmation("#confirm-modal");
+deleteConfirmPopup.setEventListeners();
+
+// Handle Delete Click
+function handleDeleteClick(cardElement) {
+  deleteConfirmPopup.open();
+
+  deleteConfirmPopup.setSubmitAction(() => {
+    deleteConfirmPopup.setLoading(true);
+    api
+      .fetchDeleteCard(cardElement.getId())
+      .then(() => {
+        cardElement.removeCard();
+        deleteConfirmPopup.close();
+      })
+      .catch(console.error)
+      .finally(() => deleteConfirmPopup.setLoading(false));
+  });
+}
+
+// Handle Like Click
+function handleLikeClick(cardElement) {
+  if (cardElement.getLikes()) {
+    api
+      .fetchDisLikeCard(cardElement.getId())
+      .then((res) => {
+        cardElement.renderLikes(res.isLiked);
+        cardElement._isLiked = res.isLiked;
+      })
+      .catch(console.error);
+  } else {
+    api
+      .fetchLikeCard(cardElement.getId())
+      .then((res) => {
+        cardElement.renderLikes(res.isLiked);
+        cardElement._isLiked = res.isLiked;
+      })
+      .catch(console.error);
+  }
+}
+
+// Initialize Form Validators
+const editFormValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```

Reply:
This is great! Thank you for this feedback. Will this be something I would see towards the end of a project or something I should do from the beginning?